src/app/labeling-suite/helper/manager/recordRla.ts

```diff
@@ -0,0 +1,203 @@
+import { getLabelSourceOrder, InformationSourceReturnType, informationSourceTypeToString, LabelSource, labelSourceToString, UserRole } from "src/app/base/enum/graphql-enums";
+import { jsonCopy } from "src/app/util/helper-functions";
+import { TableDisplayData } from "../../sub-components/overview-table/helper";
+import { getHoverGroupsOverviewTable } from "../util-functions";
+import { LabelingSuiteManager } from "./manager";
+import { GOLD_STAR_USER_ID } from "./user";
+
+export class LabelingSuiteRlaPreparator {
+
+    private rlas: any[];
+    private baseManager: LabelingSuiteManager;
+
+    constructor(baseManager: LabelingSuiteManager) {
+        this.baseManager = baseManager;
+    }
+
+    public setRlas(rlas: any, tokenAttribute: any) {
+        if (!rlas) {
+            this.rlas = null;
+            return;
+        }
+        const rlaData = this.prepareRLADataForRole(jsonCopy(rlas));
+        this.rlas = this.finalizeRlas(rlaData, tokenAttribute);
+    }
+
+    public rlasLoaded(): boolean {
+        return this.rlas != null;
+    }
+
+
+    private finalizeRlas(rlas: any[], tokenAttribute: any): any[] {
+        if (!rlas) return [];
+        for (const e of rlas) {
+            if (e.returnType == InformationSourceReturnType.RETURN) continue;
+            const attributeId = e.labelingTaskLabel.labelingTask.attribute.id;
+            let att = this.getTokenizedAttribute(attributeId, tokenAttribute);
+            let t1 = this.getToken(att, e.tokenStartIdx);
+            let t2 = this.getToken(att, e.tokenEndIdx);
+            e.value = att.raw.substring(t1.posStart, t2.posEnd);
+        }
+        return rlas;
+    }
+
+
+    private prepareRLADataForRole(rlaData: any[]): any[] {
+        if (this.baseManager.userManager.currentRole != UserRole.ANNOTATOR) return rlaData;
+        const currentSourceId = this.baseManager.sessionManager.getSourceId();
+        const userId = this.baseManager.userManager.displayUserId;
+        const allowedTask = this.baseManager.sessionManager.getAllowedTaskId();
+        rlaData.forEach((rla) => {
+            if ((rla.sourceId && rla.sourceId == currentSourceId)) {
+                rla.sourceType = LabelSource.MANUAL;
+                rla.sourceId = null;
+            } else if (this.baseManager.userManager.roleAssumed && rla.createdBy == userId && rla.labelingTaskLabel.labelingTask.id == allowedTask
+                && rla.sourceType == LabelSource.MANUAL) {
+
+            } else {
+                rla.id = "x";
+            }
+        });
```

Comment:
is intential -> however I prefer this version since the else case is imo easier to read like this "remove all others" is the final idea => ill add a comment to reduce confusion though :)